src/test/fuzz/base_encode_decode.cpp

```diff
@@ -19,42 +20,40 @@ using util::TrimStringView;
 
 FUZZ_TARGET(base58_encode_decode)
 {
-    FuzzedDataProvider provider(buffer.data(), buffer.size());
-    const std::string random_string{provider.ConsumeRandomLengthString(1000)};
-    const int max_ret_len{provider.ConsumeIntegralInRange<int>(-1, 1000)};
+    FuzzedDataProvider provider{buffer.data(), buffer.size()};
+    const auto random_string{provider.ConsumeRandomLengthString(100)};
+    const int max_ret_len{provider.ConsumeIntegralInRange<int>(-1, 100)};
 
     // Decode/Encode roundtrip
-    std::vector<unsigned char> decoded;
-    if (DecodeBase58(random_string, decoded, max_ret_len)) {
+    if (std::vector<unsigned char> decoded; DecodeBase58(random_string, decoded, max_ret_len)) {
         const auto encoded_string{EncodeBase58(decoded)};
         assert(encoded_string == TrimStringView(random_string));
-        assert(encoded_string.empty() || !DecodeBase58(encoded_string, decoded, provider.ConsumeIntegralInRange<int>(0, decoded.size() - 1)));
+        assert(decoded.empty() || !DecodeBase58(encoded_string, decoded, provider.ConsumeIntegralInRange<int>(0, decoded.size() - 1)));
```

Comment:
why is the check changed from `encoded_string.empty()` to `decoded.empty()` in the first commit - especially if this function is being rewritten in the next commit anyway?

Reply:
it's the decoded size that determines if the `0, decoded.size() - 1` interval makes sense or not - I'm correcting it in the first commit since this was a bug that could cause a fuzz failure (added explanation to commit message).
I'm reordering in the second commit so that we have single roundtrip logic instead of 2 separate ones.